smartchr.el

```diff
@@ -91,79 +100,20 @@
          :cleanup-fn (lambda ())
          :insert-fn (lambda ()))))))
    ((string-match smartchr-template-cursor-re template)
-    (destructuring-bind (pre post) (split-string template smartchr-template-cursor-re)
-      (lexical-let ((pre pre) (post post))
+    (cl-destructuring-bind (pre post) (split-string template smartchr-template-cursor-re)
+      (let ((pre pre) (post post))
         (smartchr-make-struct
          :cleanup-fn (lambda ()
-                       (delete-backward-char (length pre))
-                       (delete-backward-char (- (length post))))
+                       (delete-char (- (length pre)))
```

Comment:
> Use `delete-char` instead of `delete-backward-char`

The reason for this is that you thought delete-char was easier to understand?

I agree with the change, but I wanted to know why.

Reply:
@zonuexe If possible, please let me know.

Reply:
Oh! Understood! Thank you!!!